tests/src/junit/java/net/neoforged/neoforge/unittest/transfer/InstancedHandlerTests.java

```diff
@@ -0,0 +1,115 @@
+        Assertions.assertThat(handler.getCapacity(0, emptyResource)).withFailMessage("Voids should hava capacity of max int").isEqualTo(Integer.MAX_VALUE);
+        Assertions.assertThat(handler.getResource(0)).withFailMessage("Empty resource should match").isEqualTo(emptyResource);
+
+        try (var transaction = TransactionManager.open(null)) {
+            Assertions.assertThat(Integer.MAX_VALUE)
+                    .withFailMessage("Void should be able to accept infinite resources")
+                    .isEqualTo(handler.insert(0, emptyResource, Integer.MAX_VALUE, transaction))
+                    .isEqualTo(handler.insert(emptyResource, Integer.MAX_VALUE, transaction));
+
+            Assertions.assertThat(0)
+                    .withFailMessage("Void should be provide no resources")
+                    .isEqualTo(handler.extract(0, emptyResource, Integer.MAX_VALUE, transaction))
+                    .isEqualTo(handler.extract(emptyResource, Integer.MAX_VALUE, transaction));
+        }
+    }
+
+    private static <T extends IResource> void testEndlessResource(T resource, T emptyResource) {
+        InfiniteResourceHandler<T> handler = new InfiniteResourceHandler<>(resource);
+        Assertions.assertThat(handler.size()).withFailMessage("Size should be 1").isEqualTo(1);
+        Assertions.assertThat(handler.supportsExtraction()).withFailMessage("Extraction should be allowed").isTrue().isEqualTo(handler.supportsExtraction(0));
+        Assertions.assertThat(handler.supportsInsertion()).withFailMessage("Insertion should not be allowed").isFalse().isEqualTo(handler.supportsInsertion(0));
+        Assertions.assertThat(ResourceHandlerUtil.isValid(handler, resource)).withFailMessage("Endless resource can not be inserted into, thus isValid should be false").isFalse();
+        Assertions.assertThat(Integer.MAX_VALUE).withFailMessage("Capacity should be max int")
+                .isEqualTo(handler.getCapacity(0, resource))
+                .isEqualTo(handler.getCapacity(0, emptyResource));
+
+        Assertions.assertThat(resource).withFailMessage("Resources should match").isEqualTo(handler.getResource(0));
+
+        try (var transaction = TransactionManager.open(null)) {
+            Assertions.assertThat(0)
+                    .withFailMessage("InfiniteHandler shouldn't be insertable into")
+                    .isEqualTo(handler.insert(0, resource, 1, transaction))
+                    .isEqualTo(handler.insert(resource, 1, transaction));
+
+            Assertions.assertThat(1337)
+                    .withFailMessage("InfiniteHandler should provide as much as we desire")
+                    .isEqualTo(handler.extract(0, resource, 1337, transaction))
+                    .isEqualTo(handler.extract(resource, 1337, transaction));
+        }
+    }
+
+    private static <T extends IResource> void testEmptyHandler(EmptyResourceHandler<T> handler, T emptyResource) {
```

Comment:
Test that the methods that don't use an index (namely the insert, extract, and supports), don't throw.